src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1527,12 +1527,12 @@ public void scheduleFeedbackSessionPublishedEmails() {
     }
 
     public void deleteFeedbackSessionsForCourse(String courseId) {
-        List<FeedbackSessionAttributes> sessionsToDelete =
-                fsDb.getFeedbackSessionsForCourse(courseId);
-
-        for (FeedbackSessionAttributes session : sessionsToDelete) {
-            deleteFeedbackSessionCascade(session.feedbackSessionName, session.courseId);
-        }
+        frcLogic.deleteFeedbackResponseCommentsForCourse(courseId);
+        frLogic.deleteFeedbackResponsesForCourse(courseId);
+        fqLogic.deleteFeedbackQuestionsForCourse(courseId);
+        
+        List<FeedbackSessionAttributes> fsList = getFeedbackSessionsForCourse(courseId);
+        fsDb.deleteEntities(fsList);
```

Comment:
Maybe renaming/wrapping deleteEntities would make what it does clearer when called from outside fsDb.